daemon/api_interfaces_test.go

```diff
@@ -340,6 +340,66 @@ func (s *interfacesSuite) TestConnectPlugFailureNoSuchSlot(c *check.C) {
 	c.Assert(ifaces.Connections, check.HasLen, 0)
 }
 
+func (s *interfacesSuite) testConnectFailureNoSnap(c *check.C, consumer, producer bool) {
```

Comment:
personally I think this pattern is a bit confusing and having multiple boolean arguments gets hard to keep track of. I like having the Assert to make sure that it's used properly, but I think maybe we could do something like this instead:

```suggestion
func (s *interfacesSuite) testConnectFailureNoSnap(c *check.C, installedSnap string) {
    c.Assert(installedSnap == "consumer" || installedSnap == "producer", check.Equals, true, check.Commentf("installed snap must be consumer or producer"))
    consumer := installedSnap == "consumer"
    producer := installedSnap == "producer"
```

then this makes the tests easier to immediately read:

```

```

Reply:
(sorry hit enter too fast, edited the comment)

Reply:
Nice idea, thanks!